Services/Masterdata/Services.Mastedata.Web/Controllers/MasterdataController.cs

```diff
@@ -0,0 +1,90 @@
+using Lens.Services.Masterdata.Models;
+using Lens.Services.Masterdata.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Services.Masterdata.Web.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+[AllowAnonymous]
+public class MasterdataController : ControllerBase
+{
+
+    private readonly IMasterdataService _masterdataService;
+
+    public MasterdataController(IMasterdataService masterdataService)
+    {
+        _masterdataService = masterdataService;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult> Get()
+    {
+        var result = await _masterdataService.GetMasterdataTypes();
+        return Ok(result);
+    }
+
+    [HttpGet("{masterdataType}")]
+    public async Task<ActionResult> Get(string masterdataType)
+    {
+        var result = await _masterdataService.GetMasterdata(masterdataType);
+        return Ok(result);
+    }
+
+    [HttpGet("type/{id}")]
+    public async Task<ActionResult> GetMasterdataType(Guid id)
```

Comment:
the method can return Task<MasterdataTypeModel> and use return await _masterdataService.GetMasterdataType(id) without Ok(result)

Reply:
Changed